### DISCUSSION TITLE
Connect to host terminal via Theia's terminal.

### DISCUSSION THREAD

**Discussion by srikanthkumbham** [eclipse-theia/theia#9029] [Q&A]
I'm facing issues while connecting Theia's terminal to the host machine bash process. I'm launching theia using the following command from my mac machine:
```
docker run -it --init -p 3000:3000 -v "/:/home/project:cached" -e THEIA_SHELL=/bin/bash theiaide/theia:next 
```
By running theia above command, I'm expecting the terminal inside theia will behave the same way as mac's(host) terminal. But seems like I'm wrong. You could see `xpath` command is present in my host machine but not present in theia's terminal.

Host Terminal:
![Screenshot 2021-02-04 at 6 56 46 PM](https://user-images.githubusercontent.com/49729728/106983562-f4467180-671a-11eb-8742-9d766821b2b3.png)

Theia Terminal:
<img width="1021" alt="Screenshot 2021-02-04 at 6 56 41 PM" src="https://user-images.githubusercontent.com/49729728/106983577-f9a3bc00-671a-11eb-8709-5606a9cc588f.png">

Could someone please help me with this?

Thanks!

**Comment by srikanthkumbham** [comment]
@vince-fugnitto @svenefftinge any suggestions on this? thanks!

**Reply by vince-fugnitto** [reply]
@srikanthkumbham I'm not a docker expert, but I believe the assumption or expectation that the host machine's tools and applications be available by default in the docker container is incorrect. While you can try to mount your entire host machine's filesystem, I'm not certain it'll work and will likely give you permission issues.

**Reply by srikanthkumbham** [reply]
@vince-fugnitto Yeah, we cannot directly access the host machine's tools/resources. I'm wondering if we could use the `THEIA_SHELL` environment variable to connect the terminal to a remote bash process that resides outside of the container?

**Reply by vince-fugnitto** [reply]
> @vince-fugnitto Yeah, we cannot directly access the host machine's tools/resources. I'm wondering if we could use the `THEIA_SHELL` environment variable to connect the terminal to a remote bash process that resides outside of the container?

@srikanthkumbham unfortunately, I'm not sure that it'll work to try and use the environment variable to reference a host machine bash process outside the docker container without some additional work (possible [reference](https://docs.docker.com/docker-for-mac/networking/#use-cases-and-workarounds)). 

In any case, I fear I do not have the appropriate knowledge or expertise to try and get the container to interact with a remote bash process outside the container. It may be worthwhile to ask on general docker forums if it is even possible, and if so which steps to take to accomplish the use-case.

**Comment by xniti3x** [comment]
workaround:
run theia in docker and connect via ssh to host
maker sure the host has openssh installed
> sudo apt-get install openssh-server
> sudo service ssh start
get ipadress of host for exampel 192.168.2.xxx
get name of hostmashine  (see host terminal)
in theia terminal run:
ssh hostname@192.168.2.xxx
prommt for authentication and fingerprint just type "yes"
done!
